src/model_signing/_cli.py

```diff
@@ -0,0 +1,451 @@
+) -> None:
+    """Sign using a private key (paired with a public one).
+
+    Signing the model at MODEL_PATH, produces the signature at SIGNATURE_PATH
+    (as per `--signature` option).
+
+    Traditionally, signing could be achieved by using a public/private key pair.
+    Pass the signing key using `--private_key`.
+
+    Note that this method does not provide a way to tie to the identity of the
+    signer, outside of pairing the keys. Also note that we don't offer key
+    management protocols.
+    """
+    signer = in_toto_signature.IntotoSigner(
+        # TODO: The API needs clean up, it uses str, not pathlib.Path
+        key.ECKeySigner.from_path(private_key_path=private_key.as_posix())
+    )
+    _serialize_and_sign(model_path, signer, signature)
+
+
+@_sign.command(name="certificate")
+@_model_path_argument
+@_write_signature_option
+@_private_key_option
+@click.option(
+    "--signing_certificate",
+    type=pathlib.Path,
+    metavar="CERTIFICATE_PATH",
+    required=True,
+    help="Path to the signing certificate, as a PEM-encoded file.",
+)
+@_certificate_root_of_trust_option
+def _sign_certificate(
+    model_path: pathlib.Path,
+    signature: pathlib.Path,
+    private_key: pathlib.Path,
+    signing_certificate: pathlib.Path,
+    certificate_chain: list[pathlib.Path],
+) -> None:
+    """Sign using a certificate.
+
+    Signing the model at MODEL_PATH, produces the signature at SIGNATURE_PATH
+    (as per `--signature` option).
+
+    Traditionally, signing can be achieved by using keys from a certificate.
+    The certificate can also provide the identity of the signer, making this
+    method more informative than just using a public/private key pair for
+    signing.  Pass the private signing key using `--private_key` and signing
+    certificate via `--signing_certificate`. Optionally, pass a certificate
+    chain via `--certificate_chain` to establish root of trust (this option can
+    be repeated as needed, or all cerificates could be placed in a single file).
+
+    Note that we don't offer certificate and key management protocols.
+    """
+    if not certificate_chain:
+        raise click.BadOptionUsage(
+            "--certificate_chain", "No certificate chain was provided"
+        )
+    signer = in_toto_signature.IntotoSigner(
+        # TODO: The API needs clean up, it uses str, not pathlib.Path
+        pki.PKISigner.from_path(
+            private_key.as_posix(),
+            signing_certificate.as_posix(),
+            [c.as_posix() for c in certificate_chain],
+        )
+    )
+    _serialize_and_sign(model_path, signer, signature)
+
+
+def _serialize_and_sign(
+    model_path: pathlib.Path, signer: signing.Signer, signature: pathlib.Path
+) -> None:
+    """Serialize a model and sign it with the provided signer."""
+
+    def hasher_factory(file_path: pathlib.Path) -> file.FileHasher:
+        return file.SimpleFileHasher(
+            file=file_path, content_hasher=memory.SHA256()
+        )
```

Comment:
do we want this sort of thing customizable? How closely do we want the CLI script to mirror whats available in `model_signing.api`?

Reply:
Yeah, we should customize it more and just use the higher level API.

For this PR I wanted to have the same CLI abilities and next we can work on migrating this to use the higher level API and have the higher level API support all configuration we support.